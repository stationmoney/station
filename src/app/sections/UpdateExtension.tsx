import classNames from "classnames"
import { EXTENSION } from "config/constants"
import styles from "./UpdateExtension.module.scss"

const UpdateExtension = () => {
  console.log(
    "window.isTerraExtensionAvailable",
    window.isTerraExtensionAvailable
  )
  console.log(
    "window.isStationExtensionAvailable",
    window.isStationExtensionAvailable
  )
  if (window.isTerraExtensionAvailable && !window.isStationExtensionAvailable)
    return (
      <div className={classNames(styles.text, styles.info)}>
        Please{" "}
        <a href={EXTENSION} target="_blank" rel="noreferrer">
          update
        </a>{" "}
        your version of Station Extension.
      </div>
    )

  return null
}

export default UpdateExtension
