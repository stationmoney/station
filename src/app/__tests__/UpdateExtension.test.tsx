import React from "react"
import { render } from "@testing-library/react"
import UpdateExtension from "../sections/UpdateExtension"

describe("UpdateExtension component", () => {
  beforeEach(() => {
    window.isTerraExtensionAvailable = true
    window.isStationExtensionAvailable = false
  })
  it("should match snapshot when Terra extension is available and Station extension is not available", () => {
    const { asFragment } = render(<UpdateExtension />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("should match snapshot when Terra extension is not available and Station extension is not available", () => {
    window.isTerraExtensionAvailable = false

    const { asFragment } = render(<UpdateExtension />)

    expect(asFragment()).toMatchSnapshot()
  })

  it("should match snapshot when Terra extension is not available and Station extension is available", () => {
    window.isTerraExtensionAvailable = false
    window.isStationExtensionAvailable = true

    const { asFragment } = render(<UpdateExtension />)

    expect(asFragment()).toMatchSnapshot()
  })
})
