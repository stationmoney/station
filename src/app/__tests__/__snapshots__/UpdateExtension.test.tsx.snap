// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateExtension component should match snapshot when Terra extension is available and Station extension is not available 1`] = `
<DocumentFragment>
  <div
    class="text info"
  >
    Please 
    <a
      href="https://chrome.google.com/webstore/detail/aiifbnbfobpmeekipheeijimdpnlpgpp"
      rel="noreferrer"
      target="_blank"
    >
      update
    </a>
     your version of Station Extension.
  </div>
</DocumentFragment>
`;

exports[`UpdateExtension component should match snapshot when Terra extension is not available and Station extension is available 1`] = `<DocumentFragment />`;

exports[`UpdateExtension component should match snapshot when Terra extension is not available and Station extension is not available 1`] = `<DocumentFragment />`;
